db/migrate/20191231000858_migrate_tags.rb

```diff
@@ -0,0 +1,109 @@
+class MigrateTags < ActiveRecord::Migration[5.2]
+  include ActiveSupport::Inflector
+
+  def up
+    create_table :taggings do |t|
+      t.references :tag, type: :int
+      t.references :taggable, polymorphic: true, type: :int
+      t.references :tagger, polymorphic: true, type: :int
+
+      t.string :context, limit: 128
+
+      t.datetime :created_at
+    end
+
+    add_column :tags, :taggings_count, :integer, default: 0
+
+    Setting.where(owned: true).each do |tag|
+      ActsAsTaggableOn::Tagging.create!(tag: tag, taggable_id: tag.user_id, taggable_type: 'User', context: 'settings')
+    end
+
+    GalleryGroup.all.each do |group|
+      users = group.gallery_tags.joins(:gallery).select(:user_id).distinct.pluck(:user_id)
+      users.each { |user_id| ActsAsTaggableOn::Tagging.create!(tag: group, taggable_id: user_id, taggable_type: 'User', context: 'gallery_groups') }
+    end
+
+    PostTag.all.each { |tagging| create_tagging(tagging, Post) }
+    CharacterTag.all.each { |tagging| create_tagging(tagging, Character) }
+    GalleryTag.all.each { |tagging| create_tagging(tagging, Gallery) }
+    TagTag.all.each { |tagging| create_tagging(tagging, Setting) }
+
+    drop_table :post_tags
+    drop_table :character_tags
+    drop_table :gallery_tags
+    drop_table :tag_tags
+    remove_column :tags, :user_id
+    remove_column :tags, :owned
+
+    ActsAsTaggableOn::Tag.reset_column_information
+    ActsAsTaggableOn::Tag.find_each do |tag|
+      ActsAsTaggableOn::Tag.reset_counters(tag.id, :taggings)
+    end
```

Comment:
On consideration this basically serves no purpose for us and should probably go.